src/main/scala/com/monsanto/arch/cloudformation/model/resource/Redshift.scala

```diff
@@ -0,0 +1,138 @@
+    DBName: Token[String],
+    ElasticIp: Option[ResourceRef[`AWS::EC2::EIP`]] = None,
+    Encrypted: Option[Boolean] = None,
+    HsmClientCertificateIdentifier: Option[Token[String]] = None,
+    HsmConfigurationIdentifier: Option[Token[String]] = None,
+    MasterUsername: Token[String],
+    MasterUserPassword: Token[String],
+    NodeType: Token[String],
+    NumberOfNodes: Option[Token[Int]], // required if multi-node
+    OwnerAccount: Option[Token[String]] = None, // required if restoring snapshot
+    Port: Option[Int] = None,
+    PreferredMaintenanceWindow: Option[Token[String]] = None,
+    PubliclyAccessible: Option[Boolean] = None,
+    SnapshotClusterIdentifier: Option[Token[String]] = None,
+    SnapshotIdentifier: Option[Token[String]] = None,
+    VpcSecurityGroupIds: Option[Seq[ResourceRef[`AWS::EC2::SecurityGroup`]]] = None,
+    override val Condition: Option[ConditionRef] = None)
+  extends Resource[`AWS::Redshift::Cluster`] {
+  def when(newCondition: Option[ConditionRef] = Condition) = copy(Condition = newCondition)
+}
+/** Sorry this looks ugly but arity only goes to 22 :( */
+object `AWS::Redshift::Cluster` extends DefaultJsonProtocol {
+  implicit val format: JsonFormat[`AWS::Redshift::Cluster`] = new RootJsonFormat[`AWS::Redshift::Cluster`] {
+    def write(c: `AWS::Redshift::Cluster`) = {
+      val obj = JsObject(
+        "AllowVersionUpgrade" -> c.AllowVersionUpgrade.toJson,
+        "AutomatedSnapshotRetentionPeriod" -> c.AutomatedSnapshotRetentionPeriod.toJson,
+        "AvailabilityZone" -> c.AvailabilityZone.toJson,
+        "ClusterParameterGroupName" -> c.ClusterParameterGroupName.toJson,
+        "ClusterSecurityGroups" -> c.ClusterSecurityGroups.toJson,
+        "ClusterSubnetGroupName" -> c.ClusterSubnetGroupName.toJson,
+        "ClusterType" -> c.ClusterType.toJson,
+        "ClusterVersion" -> c.ClusterVersion.toJson,
+        "DBName" -> c.DBName.toJson,
+        "ElasticIp" -> c.ElasticIp.toJson,
+        "Encrypted" -> c.Encrypted.toJson,
+        "HsmClientCertificateIdentifier" -> c.HsmClientCertificateIdentifier.toJson,
+        "HsmConfigurationIdentifier" -> c.HsmConfigurationIdentifier.toJson,
+        "MasterUsername" -> c.MasterUsername.toJson,
+        "MasterUserPassword" -> c.MasterUserPassword.toJson,
+        "NodeType" -> c.NodeType.toJson,
+        "NumberOfNodes" -> c.NumberOfNodes.toJson,
+        "OwnerAccount" -> c.OwnerAccount.toJson,
+        "Port" -> c.Port.toJson,
+        "PreferredMaintenanceWindow" -> c.PreferredMaintenanceWindow.toJson,
+        "PubliclyAccessible" -> c.PubliclyAccessible.toJson,
+        "SnapshotClusterIdentifier" -> c.SnapshotClusterIdentifier.toJson,
+        "SnapshotIdentifier" -> c.SnapshotIdentifier.toJson,
+        "VpcSecurityGroupIds" -> c.VpcSecurityGroupIds.toJson
+      )
+      obj.copy(fields = obj.fields.filter(_._2 != JsNull))
+    }
+
+    // TODO: does anyone use this code to read JSON in rather than just write?
```

Comment:
not yet, but if you want to have a Seq[R <: Resource[R]] the Spray-JSON type class for Seq[T : JsonFormat] needs a whole JsonFormat and not just a writer, so that's why we can these fake impls all over now.